src/app/zap-templates/partials/cluster-objects-struct.zapt

```diff
@@ -8,13 +8,17 @@ namespace {{asUpperCamelCase name}} {
 
     struct Type {
     public:
+    {{#if struct_contains_array}}
         {{#zcl_struct_items}}
-        {{zapTypeToEncodableClusterObjectType type cluster=../cluster}} {{asLowerCamelCase label}}{{> cluster_objects_field_init cluster=../cluster}};
+        {{zapTypeToDecodableClusterObjectType type cluster=../cluster}} {{asLowerCamelCase label}}{{> cluster_objects_field_init cluster=../cluster}};
```

Comment:
This needs a comment at least, explaining why an encode-only type is using zapTypeToDecodableClusterObjectType.

Reply:
It's not an encode-only type; the TL;DR of the zapt change is that instead of generating 2 types, `Type` & `DecodableType` when a a struct has an array, we always generate a single type, regardless (notice the `DecodableType` code a few line below deleted)

This matches the behavior for when a struct doesn't have an array, where `using DecodableType = Type;` - that is now always the case, whether the struct has an array or not

Reply:
No longer applies with the shift to not combine Type + DecodableType (code removed)